### PR TITLE
Honda: Car Port for Clarity 2018-22

### PR DESCRIPTION
**Pull request by mcallbosco**
Hi, I'm Mcall#1636 from discord. This pull request should allow the Honda clarity to work with panda safety. 

I did some testing today and these changes seem to work properly, and I do not see any reason why they shouldn't work. I will test this specific branch again in a few days and mark this as ready when I do that.

I did want to use a console output for debugging, just to 100% make sure that it is working properly, but it does not seem that I can include iostream. Any tips? Since this is panda safety code, it doesn't seem to affect normal OP functionality in a way where I can easily test it. I do not know why it wouldn't work or falsely activate, since it follows the same structure as the other safety params.

### CODE REVIEW

**Comment by sunnyhaibin** [comment]
You will need to access the debug console for panda, or [comma serial](https://docs.comma.ai/tools/serial/README.html). I have never used them before so I can't comment too much on it.

If the only modification is a brake signal check, and is set properly on the openpilot side either via FW fingerprinting or CAN fingerprinting on initialization, a route with raw logs uploaded to show the signal is working should be good.

**Comment by mrnerdhair** [comment]
Another proud Clarity owner here, looking forward to this! Happy to test something if it would help.

**Comment by sunnyhaibin** [comment]
@mcallbosco Can you attach a route with raw logs uploaded for verification? Were there any changes on the openpilot/opendbc to work?

**Comment by mcallbosco** [comment]
Route with param/messages set incorrectly in the panda for the clarity: 5b0bcddfca0b8e10|2023-04-06--13-18-31

Route with param/messages set correctly in the panda for the clarity: 5b0bcddfca0b8e10|2023-04-06--13-34-02

File: board/safety/safety_honda.h [hunk 1]
```diff
@@ -369,7 +376,14 @@ static const addr_checks* honda_nidec_init(uint16_t param) {
...
   honda_bosch_radarless = false;
-
+ 
+  
+  if (GET_FLAG(param, HONDA_PARAM_CLARITY)){
+    honda_clarity_brake_msg = true;
+    print("Honda Clarity Brake Messages Active\n");
+  } else {
+    honda_clarity_brake_msg = false;
+  }
```
Comment (sunnyhaibin) [inline board/safety/safety_honda.h:386]:
```suggestion
  honda_clarity_brake_msg = GET_FLAG(param, HONDA_PARAM_CLARITY);
```

Comment (sunnyhaibin) [reply board/safety/safety_honda.h:386]: We can simplify this block as suggested.

File: board/safety/safety_honda.h [hunk 2]
```diff
@@ -369,7 +376,7 @@ static const addr_checks* honda_nidec_init(uint16_t param) {
   honda_alt_brake_msg = false;
   honda_bosch_long = false;
   honda_bosch_radarless = false;
-
+  honda_clarity_brake_msg = GET_FLAG(param, HONDA_PARAM_CLARITY);
```
Comment (sunnyhaibin) [inline board/safety/safety_honda.h:379]:
```suggestion

  honda_clarity_brake_msg = GET_FLAG(param, HONDA_PARAM_CLARITY);
```

File: board/safety/safety_honda.h [hunk 3]
```diff
@@ -370,6 +377,7 @@ static const addr_checks* honda_nidec_init(uint16_t param) {
   honda_bosch_long = false;
   honda_bosch_radarless = false;
 
+  honda_clarity_brake_msg = GET_FLAG(param, HONDA_PARAM_CLARITY);
```
Comment (sunnyhaibin) [inline board/safety/safety_honda.h:380]:
```suggestion
  honda_clarity_brake_msg = GET_FLAG(param, HONDA_PARAM_CLARITY);

```

**Comment by sunnyhaibin** [comment]
Thanks for the PR @mcallbosco! Clarity users - enjoy 😇

**Pull request merged**